pkg/stockings/stock.go

```diff
@@ -0,0 +1,159 @@
+	different apis via StockAPI interface.
+
+*/
+type Stock struct {
+	Symbol        string  `json:"symbol"`
+	Name          string  `json:"companyName"`
+	LatestPrice   float32 `json:"latestPrice"`
+	Change        float32 `json:"change"`
+	ChangePercent float32 `json:"changePercent`
+}
+
+type HistoricalStock struct {
+	Date  string  `json:"date"`
+	Price float32 `json:"close"`
+}
+
+type HistoricalStocks []HistoricalStock
+
+const (
+	iexCurrentPriceUrl   = "https://cloud.iexapis.com/stable/stock/%s/quote?token=%s"
+	iexHistoricalDateUrl = "https://cloud.iexapis.com/stable/stock/%s/chart/%s?token=%s"
+	dateLayout           = "20060102"
+	iexDateLayout        = "2006-01-02"
+)
+
+//example for ralles, he should refactor this to better handle error checking etc
+//since this is a large struct, should we perhaps return *Stock?
+func GetCurrentPrice(ticker string) (Stock, error) {
+	url := fmt.Sprintf(iexCurrentPriceUrl, ticker, os.Getenv("IEX_TOKEN"))
+	resp, err := http.Get(url)
+	if err != nil {
+		return Stock{}, err
+	}
+	if resp.StatusCode != 200 {
+		return Stock{}, errors.New("http resp not 200")
+	}
+	var quote Stock
+	err = json.NewDecoder(resp.Body).Decode(&quote)
+	if err != nil {
+		return Stock{}, err
+	}
+	return quote, nil
+}
+
+//function that returns HistoricalStock at a certain date
+func GetHistoricalPrice(ticker string, date string) (HistoricalStock, error) {
+	url := fmt.Sprintf(iexHistoricalDateUrl, ticker, "date/"+date+"?chartByDay=true", os.Getenv("IEX_TOKEN"))
+	resp, err := http.Get(url)
+	if resp.StatusCode != 200 {
+		return HistoricalStock{}, errors.New("http resp not 200")
+	}
+	var historical HistoricalStocks
+	err = json.NewDecoder(resp.Body).Decode(&historical)
+	if err != nil {
+		return HistoricalStock{}, err
+	}
+	if len(historical) != 1 {
+		return HistoricalStock{}, errors.New("did not return singular value after unmarshall")
+	}
+	return historical[0], nil
+}
+
+//function that returns a pointer to a slice of HistoricalStock's for a date range
+//do we need to return *[]*HistoricalStock?
+func GetHistoricalRange(ticker string, start string, end string) (*HistoricalStocks, error) {
+	rangeQuery, err := getRange(start)
+	if err != nil {
+		return nil, err
+	}
+	url := fmt.Sprintf(iexHistoricalDateUrl, ticker, rangeQuery, os.Getenv("IEX_TOKEN"))
+	resp, err := http.Get(url)
+	if resp.StatusCode != 200 {
+		return nil, errors.New("http resp not 200")
+	}
+	var historical HistoricalStocks
+	err = json.NewDecoder(resp.Body).Decode(&historical)
+	if err != nil {
+		return nil, err
+	}
+	//this is how this would be done in c++...modify param since it's a pointer
+	err = parseHistoricalRange(&historical, start, end)
+	if err != nil {
+		return nil, err
+	}
+	return &historical, nil
+}
+
+//taken from https://github.com/addisonlynch/iexfinance/blob/master/iexfinance/stocks/historical.py
+//some leap year stuff is fucked up for them so will have to rewrite this later probably
+func getRange(date string) (string, error) {
```

Comment:
I feel like I see this for a lot of the code in the PR - usually I think you want to return `(pointer, error)` instead of `(value, error)`. so in this case, should return `(*string, error)`, and replace `""` with `nil`

Reply:
alright so returning pointers is just standard in go im assuming

Reply:
resolved.